reframe/utility/__init__.py

```diff
@@ -1473,3 +1473,11 @@ def __repr__(self):
 
     def __str__(self):
         return str(self.__mapping)
+
+    def __add__(self, other):
+        if not isinstance(other, type(self)):
+            return NotImplemented
+
+        new_mapping = self.__mapping.copy()
+        new_mapping.update(other.__mapping)
+        return MappingView(new_mapping)
```

Comment:
I don't think that it's a right approach to support `__add__` either here or in the `SequenceView`. Here we're offering a view to something that is not related to the original container. This has come as a workaround for the flex allocation node, but I don't think it's the right way. I believe that these view should provide access to the underlying container with a `data()` method.